### DISCUSSION TITLE
seems the response textDocument/hover is incomplete compare to  official document

### DISCUSSION THREAD

**Discussion by haolian9** [phpactor/phpactor#2003] [Q&A] [answered]
Hi,
I'm not sure if this is a problem on my neovim setup or on the phpactor side.
that's what I got in nvim, do you think it's an expected response?

```
### array_map

Applies the callback to the elements of the given arrays
https://php.net/manual/en/function.array-map.php

Callback function to run for each element in each array.

An array to run through the callback function.

after applying the callback function to each one.

<?php function array_map(callable()|null $callback, array $array, array[] $arrays): array
```

as you can see, the parameter part is missing, IMO it should be something like that: (here's [the official doc](https://www.php.net/manual/en/function.array-map))

```
callback: 
    Callback function to run for each element in each array.

array:
    An array to run through the callback function.

arrays:
    after applying the callback function to each one.
```

I do checked the response from phpactor before nvim's builtin lsp client
processing it, and the lsp client did not trim/filter these info.

**Comment by dantleech** [comment, marked as answer]
Yes this is this expected response currently, it could be improved. I've created issue #2004 for this.

**Reply by haolian9** [reply]
Thanks for confirming and considering this!

**Comment by dantleech** [comment]
![image](https://user-images.githubusercontent.com/530801/210233762-6a5ee12c-62bc-48fe-a002-063b700e72aa.png)

this should be improved now (spot the unrelated bug with the hover formatting :smile: )
